utilities/transactions/pessimistic_transaction.cc

```diff
@@ -217,6 +217,60 @@ inline Status WriteCommittedTxn::GetForUpdateImpl(
                                            value, exclusive, do_validate);
 }
 
+Status WriteCommittedTxn::GetEntityForUpdate(const ReadOptions& read_options,
+                                             ColumnFamilyHandle* column_family,
+                                             const Slice& key,
+                                             PinnableWideColumns* columns,
+                                             bool exclusive, bool do_validate) {
+  if (!column_family) {
+    return Status::InvalidArgument(
+        "Cannot call GetEntityForUpdate without a column family handle");
+  }
+
+  const Comparator* const ucmp = column_family->GetComparator();
+  assert(ucmp);
+  const size_t ts_sz = ucmp->timestamp_size();
+
+  if (ts_sz == 0) {
+    return TransactionBaseImpl::GetEntityForUpdate(
+        read_options, column_family, key, columns, exclusive, do_validate);
+  }
+
+  assert(ts_sz > 0);
+
+  if (!do_validate) {
+    if (read_timestamp_ != kMaxTxnTimestamp) {
+      return Status::InvalidArgument(
```

Comment:
Just a thought. Could it be confusing to the users if we ignore `do_validate` being false and always just force validate when `ts` is set?

Reply:
Yeah, I think silently performing validation when the application passed in `do_validate==false` might be a bit confusing, especially for users migrating from `GetForUpdate` to `GetEntityForUpdate`. So this (and the below) case is where I feel it might make sense to be consistent